Google.Api.Generator/Utils/ResourcePattern.cs

```diff
@@ -41,13 +41,21 @@ private class CollectionIdentifierSegment : Segment
         {
             public CollectionIdentifierSegment(string segment)
             {
-                bool valid = segment != "" &&
-                    segment[0] >= 'a' && segment[0] <= 'z' && segment.All(c => (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9'));
+                Identifier = segment;
+                Validate(tightValidation: false);
+            }
+
+            public void Validate(bool tightValidation)
```

Comment:
Is this ever called with tightValidation=false? I can only see the call in the ResourcePattern constructor, where it's false.

Reply:
Yes, in the `CollectionIdentifierSegment` constructor it's called with `tightValidation: false`.

Reply:
Ah, whoops. Hadn't seen that. As you were :)